src/spatial.jl

```diff
@@ -421,51 +422,110 @@ function wrap_coords(v, boundary::TriclinicBoundary)
+function random_velocity_3D(mass::Union{Unitful.Mass, MolarMass}, temp::Unitful.Temperature,
+     k::Union{BoltzmannConstUnits, MolarBoltzmannConstUnits}, rng=Random.GLOBAL_RNG)
     return SVector(
         maxwell_boltzmann(mass, temp, k; rng=rng),
         maxwell_boltzmann(mass, temp, k; rng=rng),
         maxwell_boltzmann(mass, temp, k; rng=rng),
     )
 end
 
-function random_velocity_2D(mass, temp, k=mb_conversion_factor, rng=Random.GLOBAL_RNG)
+function random_velocity_3D(mass::AbstractFloat, temp::AbstractFloat, k::AbstractFloat, rng=Random.GLOBAL_RNG)
+    return SVector(
+        maxwell_boltzmann(mass, temp, k; rng=rng),
+        maxwell_boltzmann(mass, temp, k; rng=rng),
+        maxwell_boltzmann(mass, temp, k; rng=rng),
+    )
+end
+
+function random_velocity_2D(mass::Union{Unitful.Mass, MolarMass}, temp::Unitful.Temperature, rng=Random.GLOBAL_RNG)
+    return SVector(
+        maxwell_boltzmann(mass, temp; rng=rng),
+        maxwell_boltzmann(mass, temp; rng=rng),
+    )
+end
+
+function random_velocity_2D(mass::Union{Unitful.Mass, MolarMass}, temp::Unitful.Temperature,
+     k::Union{BoltzmannConstUnits, MolarBoltzmannConstUnits}, rng=Random.GLOBAL_RNG)
+    return SVector(
+        maxwell_boltzmann(mass, temp, k; rng=rng),
+        maxwell_boltzmann(mass, temp, k; rng=rng),
+    )
+end
+
+function random_velocity_2D(mass::AbstractFloat, temp::AbstractFloat, k::AbstractFloat, rng=Random.GLOBAL_RNG)
     return SVector(
         maxwell_boltzmann(mass, temp, k; rng=rng),
         maxwell_boltzmann(mass, temp, k; rng=rng),
     )
 end
 
 """
-    maxwell_boltzmann(mass, temperature; rng=Random.GLOBAL_RNG)
+    maxwell_boltzmann(mass::Unitful.Mass, temperature::Unitful.Temperature; rng=Random.GLOBAL_RNG)
+    maxwell_boltzmann(mass::MolarMass, temperature::Unitful.Temperature; rng=Random.GLOBAL_RNG)
     maxwell_boltzmann(mass, temperature, k; rng=Random.GLOBAL_RNG)
 
 Generate a random velocity along one dimension from the Maxwell-Boltzmann
 distribution, with optional custom Boltzmann constant.
 """
-function maxwell_boltzmann(mass, temp, k; rng=Random.GLOBAL_RNG)
+function maxwell_boltzmann(mass::Unitful.Mass, temp::Unitful.Temperature; rng=Random.GLOBAL_RNG)
+    T = typeof(convert(AbstractFloat, ustrip(temp)))
+    σ = sqrt(Unitful.k * temp / mass)
+    return rand(rng, Normal(zero(T), T(ustrip(σ)))) * unit(σ)
+end
+
+function maxwell_boltzmann(mass::Unitful.Mass, temp::Unitful.Temperature, k::BoltzmannConstUnits; rng=Random.GLOBAL_RNG)
     T = typeof(convert(AbstractFloat, ustrip(temp)))
     σ = sqrt(k * temp / mass)
     return rand(rng, Normal(zero(T), T(ustrip(σ)))) * unit(σ)
 end
 
-function maxwell_boltzmann(mass, temp; rng=Random.GLOBAL_RNG)
-    k = unit(temp) == NoUnits ? ustrip(mb_conversion_factor) : mb_conversion_factor
-    return maxwell_boltzmann(mass, temp, k; rng=rng)
+function maxwell_boltzmann(mass::MolarMass, temp::Unitful.Temperature; rng=Random.GLOBAL_RNG)
```

Comment:
This could be rolled into the below method with `k_molar=Unitful.k * Unitful.Na` (also above)?